client/src/reducers/DirectoryReducer.js

```diff
@@ -12,11 +12,17 @@ const DirectoryReducer = (state = initialState, action) => {
       return {
         ...state,
         loading: true,
+        errorMessage: undefined,
       };
     case 'ADD_USER_SUCCESS':
       newState = Object.assign({}, state, { loading: false });
       newState[action.role].push(action.user);
       return newState;
+    case 'ADD_USER_FAILURE':
+      return {
+        ...state,
+        errorMessage: action.error,
```

Comment:
If you look around, there are some checks like this `if (response && !response.error)` or `if (response && response.error) return;` to check if response contains an error, with this change in naming (i.e. from `error` to `errorMessage`), the checks for `!response.error` will pass. For this reason, I prefer to keep this naming consistent across all the failure actions dispatched.

Reply:
I dont agree. An error is everything about that error, such as statuscode etc. We still dispatch it as error, and can use it as error in the promise. I can change it if you want it to be more generic

Reply:
Nah if you prefer to pass it through the App, then this is fine.